### ISSUE TITLE
Easier way to fold TaskEither?

### ISSUE THREAD

**Issue by mlegenhausen** [gcanti/fp-ts#583]
The `TaskEither.fold` function looks a little bit suprising for me. Instead that I can fold the whole monad I am forced to return a `Task` and can only fold the inner `Either`.

I expected something like this:

```ts
const fold = <L1, A, L2, B>(
  fa: taskEither.TaskEither<L1, A>,
  left: (l: L1) => taskEither.TaskEither<L2, B>,
  right: (a: A) => taskEither.TaskEither<L2, B>
) => new taskEither.TaskEither(
  fa.value.chain((a) => a.fold(
    (l) => left(l).value,
    (a) => right(a).value
  ))
);
```

Where is my error in reasoning?

**Comment by gcanti** [comment]
The semantic of `fold` is trasforming an async computation that can fail into an async computation that never fails, basically `Either.fold` lifted to `Task`. 

We can add your combinator as a method, not sure about the name though.

**Comment by mlegenhausen** [comment]
Closed in favor of #579.
